coderd/database/queries/workspaces.sql

```diff
@@ -687,3 +687,39 @@ UPDATE workspaces SET favorite = true WHERE id = @id;
 
 -- name: UnfavoriteWorkspace :exec
 UPDATE workspaces SET favorite = false WHERE id = @id;
+
+-- name: GetWorkspacesAndAgentsByOwnerID :many
+SELECT
+	workspaces.id as id,
+	workspaces.name as name,
+	job_status,
+	transition,
+	(array_agg(ROW(agent_id, agent_name)::agent_id_name_pair) FILTER (WHERE agent_id IS NOT NULL))::agent_id_name_pair[] as agents
+FROM workspaces
+LEFT JOIN LATERAL (
+	SELECT
+		workspace_id,
+		job_id,
+		transition,
+		job_status
+	FROM workspace_builds
+	JOIN provisioner_jobs ON provisioner_jobs.id = workspace_builds.job_id
+	WHERE workspace_builds.workspace_id = workspaces.id
+	ORDER BY build_number DESC
+	LIMIT 1
+) latest_build ON true
+LEFT JOIN (
+	SELECT
+		workspace_agents.id as agent_id,
+		workspace_agents.name as agent_name,
+		job_id
+	FROM workspace_resources
+	JOIN workspace_agents ON workspace_agents.resource_id = workspace_resources.id
+) resources ON resources.job_id = latest_build.job_id
+WHERE
+	-- Filter by owner_id
+	workspaces.owner_id = @owner_id :: uuid
+	AND workspaces.deleted = false
+GROUP BY workspaces.id, workspaces.name, latest_build.job_status, latest_build.job_id, latest_build.transition;
```

Comment:
If think you can stop at `GROUP BY workspaces.id` since it's unique on its own, and all the other things you have here are 1:1 with it.

Reply:
My understanding of SQL is that if you group by any column selected on, you have to group by, or aggregate, all the others, otherwise it would be unclear what should happen to the other columns if they happen to differ.

Reply:
Yeah, I guess you're right. Annoying.